### PR TITLE
Ignore other parameters in REST content-type

### PR DESCRIPTION
**Pull request by bdunogier**
> Fixes EZP-24877: Error listing subitems on PlatformUI on Firefox

Covered by unit test (as well as version).

### CODE REVIEW

File: eZ/Publish/Core/REST/Common/Input/ParsingDispatcher.php [hunk 1]
```diff
@@ -94,12 +94,18 @@ protected function parseMediaTypeVersion($mediaType)
     {
         $version = '1.0';
         $contentType = explode('; ', $mediaType);
-        if (isset($contentType[1])) {
+        if (count($contentType) > 1) {
```
Comment (andrerom) [inline eZ/Publish/Core/REST/Common/Input/ParsingDispatcher.php]: ``` if (!empty($contentType)) {```

**Comment by andrerom** [comment]
+1

**Comment by dpobel** [comment]
that does not seem to fix the issue:
```
$ curl 'http://ez.loc/api/ezp/v2/views' -H 'X-CSRF-Token: _UIfqN-m6MmFTfxn6-pu9D0iVZVvoIQx9hXgUc7iW7g' -H 'Content-Type: application/vnd.ez.api.ViewInput+json; charset=UTF-8; version=1.1' -H 'Accept: application/vnd.ez.api.View+json; version=1.1' -H 'Cookie: is_logged_in=true; eZSESSID=g7d600gvf69q5l6o58a0kaanu2' --data-binary '{"ViewInput":{"identifier":"subitemlist-2","public":false,"LocationQuery":{"Criteria":{"ParentLocationIdCriterion":2},"FacetBuilders":{},"SortClauses":{},"offset":0,"limit":10}}}'
{
    "ErrorMessage": {
        "_media-type": "application\/vnd.ez.api.ErrorMessage+json",
        "errorCode": 400,
        "errorMessage": "Bad Request",
        "errorDescription": "Missing <Query> attribute for <ViewInput>.",
        "trace": "#0 \/home\/dp\/dev\/ezplatform\/vendor\/ezsystems\/ezpublish-kernel\/eZ\/Publish\/Core\/REST\/Common\/Input\/ParsingDispatcher.php(83): eZ\\Publish\\Core\\REST\\Server\\Input\\Parser\\ViewInput->parse(Array, Object(eZ\\Publish\\Core\\REST\\Common\\Input\\ParsingDispatcher))\n#1 \/home\/dp\/dev\/ezplatform\/vendor\/ezsystems\/ezpublish-kernel\/eZ\/Publish\/Core\/REST\/Common\/Input\/Dispatcher.php(117): eZ\\Publish\\Core\\REST\\Common\\Input\\ParsingDispatcher->parse(Array, 'application\/vnd...')\n#2 \/home\/dp\/dev\/ezplatform\/vendor\/ezsystems\/ezpublish-kernel\/eZ\/Publish\/Core\/REST\/Server\/Controller\/Views.php(41): eZ\\Publish\\Core\\REST\\Common\\Input\\Dispatcher->parse(Object(eZ\\Publish\\Core\\REST\\Common\\Message))\n#3 [internal function]: eZ\\Publish\\Core\\REST\\Server\\Controller\\Views->createView(Object(Symfony\\Component\\HttpFoundation\\Request))\n#4 \/home\/dp\/dev\/ezplatform\/vendor\/symfony\/symfony\/src\/Symfony\/Component\/HttpKernel\/HttpKernel.php(147): call_user_func_array(Array, Array)\n#5 \/home\/dp\/dev\/ezplatform\/vendor\/symfony\/symfony\/src\/Symfony\/Component\/HttpKernel\/HttpKernel.php(68): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)\n#6 \/home\/dp\/dev\/ezplatform\/vendor\/symfony\/symfony\/src\/Symfony\/Component\/HttpKernel\/DependencyInjection\/ContainerAwareHttpKernel.php(69): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#7 \/home\/dp\/dev\/ezplatform\/vendor\/symfony\/symfony\/src\/Symfony\/Component\/HttpKernel\/Kernel.php(196): Symfony\\Component\\HttpKernel\\DependencyInjection\\ContainerAwareHttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#8 \/home\/dp\/dev\/ezplatform\/web\/index.php(66): Symfony\\Component\\HttpKernel\\Kernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))\n#9 {main}",
        "file": "\/home\/dp\/dev\/ezplatform\/vendor\/ezsystems\/ezpublish-kernel\/eZ\/Publish\/Core\/REST\/Server\/Input\/Parser\/ViewInput.php",
        "line": 46
    }
}
```

**Comment by bdunogier** [comment]
That's what you get when the same thing is done distinctly in two different places.

 Behat wouldn't have missed that ;)

**Comment by bdunogier** [comment]
Also fixed in the `InputDispatcher` in addition to the `InputParsingDispatcher`.

**Comment by dpobel** [comment]
+1

**Pull request merged**